quantum/matrix.c

```diff
@@ -123,6 +126,16 @@ static bool read_cols_on_row(matrix_row_t current_matrix[], uint8_t current_row)
         return true;
     }
     return false;
+
+    #else
+    uint8_t port_state = ~PINx_ADDRESS(PORTSCAN) & PORTSCAN_PINS;
```

Comment:
This will only work for AVR. TBH I don't know if it is possible to implement on ARM anyway.

Reply:
Port scanning is definitely possible on at least some STM32 chips, here's a small writeup about doing it: https://www.engineersgarage.com/stm32/accessing-ports-of-stm32-microcontroller/

You're right though, the code I have here is _explicitly_ for AVR chips. Is there something extra that I need to do to guard against someone accidentally using this on non-AVR platforms?

Reply:
Well, the idea is that features should not be limited to either AVR or ARM (especially AVR, as ARM MCUs are generally much more powerful) if it can be helped. So if it is not possible to implement for ARM, I personally would not really be in favour of merging this unless there's a significant performance gain.

Reply:
Again, it should be possible to implement on ARM, and this could likely be extended to support that, but I don't have any ARM hardware wired the correct way on-hand to try it out. 

This was also originally implemented as a custom matrix for a single board (ketch, in #11656 ), but I was told there that it would be preferred as an addition to the standard matrix code, so... here we are. I'm happy to put it back in as a custom matrix instead, I just did it this way because I was asked in the other PR to do it this way.

Reply:
I think the best solution may be to hold off until ARM compatible code is written and somebody can test it.